### ISSUE TITLE
[Feature request] Custom loss functions

### ISSUE THREAD

**Issue by prescient** [imbs-hl/ranger#368]
It would be nice if there was a way to add custom loss functions to ranger (e.g. poisson) for model training.

**Comment by mnwright** [comment]
I agree, that would be very nice. However, these loss functions would have to be written in C++. Also the split point evaluation is the most computational intensive part of the RF algorithm and is carefully optimized for speed. Unfortunately this makes it difficult to replace the splitting by a simple function. 

Maybe ranger is not the perfect package for this. An alternative might be [simpleRF](https://github.com/mnwright/simpleRF), but it is extremely slow...

**Comment by prescient** [comment]
Would it be possible to pass a custom loss function via Rcpp and pass that to ranger? I'm not sure how viable this is, but it would be a nice feature as we commonly work with count/poisson data.

**Comment by mayer79** [comment]
I am currently working count regressions, where the aim is typically to minimize Poisson deviance. Using squared loss is not recommened in these settings. What about offering the most important loss functions as option in the interface with squared loss as default for regression and Gini impurity for classification? LightGBM (gradient boosting) as well as xgboost offers Poisson, Gamma and even Tweedie (Poisson sum of Gamma random variables) for this purpose  (https://github.com/Microsoft/LightGBM/blob/master/docs/Parameters.rst). I could not spot any random forest implementation in R (outside pseudo rfs of xgboost/lightgbm) with these options. It would be very useful though.
